src/runtime/server/settings.ts

```diff
@@ -138,20 +137,21 @@ function validateGraphQLPath(path: string): string {
   return outputPath
 }
 
-export function changeSettings(oldSettings: SettingsData, newSettings: SettingsInput): SettingsData {
-  const outputSettings = { ...oldSettings, ...newSettings }
-  const playground = playgroundSettings(outputSettings.playground)
-
-  outputSettings.path = validateGraphQLPath(outputSettings.path)
+/**
+ * Mutate the settings data
+ */
+export function changeSettings(state: SettingsData, newSettings: SettingsInput): void {
+  const updatedSettings = { ...state, ...newSettings }
 
-  return {
-    ...outputSettings,
-    playground,
-  }
+  state.playground = playgroundSettings(updatedSettings.playground)
+  state.path = validateGraphQLPath(updatedSettings.path)
+  state.port = updatedSettings.port
+  state.startMessage = updatedSettings.startMessage
```

Comment:
I guess the mutation is needed to fix something?

Reply:
Well, I'm not super happy with with how we're mutating the settings (afraid we might have bad side-effects at some point) but that's how it's done in the schema component. It's directly mutating the `settings.current`, and to keep symmetry I did the same in the server component